pandas/io/parquet.py

```diff
@@ -10,7 +11,7 @@
 
 from pandas import DataFrame, get_option
 
-from pandas.io.common import get_filepath_or_buffer, is_fsspec_url, stringify_path
+from pandas.io.common import get_handle, is_fsspec_url, stringify_path
```

Comment:
parquet is difficult:

1. `FastParquetImpl` does not support file handles for writing (`PyArrowImpl` does)
2. `to_parquet` can write multiple files to a folder (a folder cannot be opened)

I kept mostly the old code but replaced the calls to `get_filepath_or_buffer`.

Reply:
can you add this as a comment here (e.g. about why we are not using get_handle in this particular io reader) (like what you wrote here but comments in the code)